data/test/scenarios/wml_tests/UnitsWML/Attacks/test_attacks_used.cfg

```diff
@@ -0,0 +1,52 @@
+#textdomain wesnoth-test
+
+#####
+# API(s) being tested: [attack]attacks_used
+##
+# Expected end state:
+# The Elvish Archer (alice) has 3/5 attacks left after attacking the Orcish Grunt (bob)
+#####
+{COMMON_KEEP_A_B_UNIT_TEST test_attacks_used (
+    [event]
+        name=prestart
+        [modify_unit]
+            [filter]
+                id=alice
+            [/filter]
+            [effect]
+                apply_to=attack
+                set_attacks_used=2
```

Comment:
Please apply this only to the melee attack, and then check that the ranged attack still only costs 1 (or the other way round).